AS5048A-Arduino

Dmitry-Chernikov opened the pull request:
Added calculation of the average median of 16 measurements ([algorithm](https://www.wikihow.com/Find-Mean,-Median,-and-Mode)).
Modernized Read and Write functions.
Added features:
//Returns the physical quantity in angular degrees, obtained from a binary 14 bit ADC number
- RotationRawToAngle
//Returns the physical quantity in angular radians, obtained from the binary 14-bit ADC number
- RotationRawToRadian
//Returns the incremental and decrementing rotation angle in the variable RotationAngle. The variable addresses are passed to the procedure.
- AbsoluteAngleRotation
//Returns the angle minutes
- GetAngularMinutes
//Returns seconds of angle
- GetAngularSeconds
//Returns the movement of the spur gear in mm
- LinearDisplacementRack
//Returns the movement of the screw in mm
- LinearMotionHelicalGear
//Function to sort ascending
- quickSort
//The function sends the NOP command and returns the contents of the register. The NOP team is a fake write to register x0000 sensor AS5048
- DummyOperNoInf
 //The procedure records the absolute value measured by the sensor AS5048, a randomly located magnet on the axis of rotation, as zero angle position
- ProgAbsolAngleZeroPosit
//Set the zero position
- setZeroPosition
//Returns the current zero position
- getZeroPosition
//Get and clear the error register and output the register value to the Serial port
- printErrors

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Merge pull request #1 from Dmitry-Chernikov/patch-2

Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Исправлены ошибки, функция сортировки qsort (array_buffer, 16, sizeof(word),AS5048A::compare); нне работает из за функции compare

Dmitry-Chernikov committed:
Добавлена  процедура быстрой сортировки

Dmitry-Chernikov committed:
Добавлена функция быстрой сортировки

Dmitry-Chernikov committed:
Исправлены ошибки по сортировки измеренных значений для медианы средней

Dmitry-Chernikov committed:
16 элементов в array_buffer

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Add files via upload

Dmitry-Chernikov committed:
Add files via upload

Dmitry-Chernikov committed:
Delete Сканер.vi

Dmitry-Chernikov committed:
Delete Сканер порта.lvproj

Dmitry-Chernikov committed:
Delete Сканер порта.lvlps

Dmitry-Chernikov committed:
Delete Сканер порта.aliases

Dmitry-Chernikov committed:
Delete Сканер порта.aliases

Dmitry-Chernikov committed:
Delete Сканер порта.lvlps

Dmitry-Chernikov committed:
Delete Сканер порта.lvproj

Dmitry-Chernikov committed:
Delete Сканер.vi

Dmitry-Chernikov committed:
Add files via upload

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

benhowes commented:
Thanks for all the hard work @Dmitry-Chernikov! I've thought on this a lot and I think unfortunately I cannot accept this PR because this project is in english rather than russian and I do not want to end up with bi-lingual comments.

I suggest you keep your fork as a russian language version, or please feel free to re-open with english comments and I'll review. 

Thanks again!

Dmitry-Chernikov commented:
Hello, well, I will translate the comments and descriptions of the library into English, the work on the library has not yet been completed, changes will be made.

Dmitry-Chernikov commented:
Hello, well, I will translate the comments and descriptions of the library into English, the work on the library has not yet been completed, changes will be made.

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov commented:
benhowes welcome comments corrected.

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Add files via upload

Dmitry-Chernikov committed:
Update README.md

Dmitry-Chernikov committed:
Delete StepMotor/StepMotor_LCD___AS5048A_23_01_2019 directory

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.h

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update AS5048A.cpp

Dmitry-Chernikov committed:
Update sketch.ino

Dmitry-Chernikov committed:
Update AS5048A.cpp

Set the default values ​​only in the header file (AS5048A.H), and in the implementation file (AS5048A.CPP), remove them.

Dmitry-Chernikov committed:
Delete lib/AS5048A/AS5048A.h

Dmitry-Chernikov committed:
Delete lib/AS5048A/AS5048A.cpp

Dmitry-Chernikov committed:
Delete src/sketch.ino

Dmitry-Chernikov committed:
Delete .gitignore

Dmitry-Chernikov committed:
Delete ino.ini

Dmitry-Chernikov committed:
Add files via upload

Dmitry-Chernikov committed:
Добавил угловое усреднение, в медианном усреднении поправел переход между 0 и 360. Обновил пример для использования библиотеки. Привёл код к стилю lowerCamelCase.

Dmitry-Chernikov committed:
добавил файл .gitignor и ino.ini

Dmitry-Chernikov committed:
в .h файле заменил на #pragma once директива препроцессора, используемая в языках программирования C и C++ для предотвращения множественного включения одного и того же заголовочного файла

Dmitry-Chernikov committed:
добавил нормализацию угла это позволяет избежать отрицательных значений и значений, превышающих полный оборотот 0 до (2\pi).
